### ISSUE TITLE
`Dataframe<T>` does not implement `ToVectorValue`

### ISSUE THREAD

**Issue by JosiahParry** [extendr/extendr#715]
Is it possible to derive `ToVectorValue` for `Dataframe<T>`? Right now, you cannot return a `Dataframe<T>` from a function. 

Here is a real use case I have with `arcgeocode` that I am building. I can parse the json using serde_json into a struct. That struct then can be made into a `Dataframe<Suggestion>`. However, in order to return it the following must happen `.as_robj().clone()`

```rust
use extendr_api::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize, IntoDataFrameRow)]
#[serde(rename_all = "camelCase")]
pub struct Suggestion {
    pub text: String,
    pub magic_key: String,
    pub is_collection: bool,
}

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct Suggestions {
    pub suggestions: Vec<Suggestion>,
}

#[extendr]
pub fn parse_suggestions(x: &str) -> Dataframe<Suggestion> {
    let sugg = serde_json::from_str::<Suggestions>(x);
    match sugg {
        Ok(s) => Dataframe::try_from_values(s.suggestions).unwrap(),
        Err(_) => Dataframe::try_from_values(Suggestions::default().suggestions).unwrap(),
    }
}
```

```json
{"suggestions":[{"text":"Bellwood Coffee, 1366 Glenwood Ave SE, Atlanta, GA, 30316, USA","magicKey":"dHA9MCN0dj02NTc3YTY2NiNsb2M9Mzk1NTg5MSNsbmc9NTUjcGw9MzIxNjg2NyNsYnM9MTQ6OTIyMjAyNSNsbj1Xb3JsZA==","isCollection":false},{"text":"Bellwood, Atlanta, GA, USA","magicKey":"dHA9MCN0dj02NTc3YTY2NiNsb2M9Mzc2MzI3MSNsbmc9NTUjcGw9MzExMzcwNyNsYnM9MTQ6OTIyMjAwNCNsbj1Xb3JsZA==","isCollection":false},{"text":"Bellwood Church, Atlanta, GA, USA","magicKey":"dHA9MCN0dj02NTc3YTY2NiNsb2M9NDIyMTM5MSNsbmc9NTUjcGw9MzQxNjY2MCNsYnM9MTQ6OTIyMjAyNCNsbj1Xb3JsZA==","isCollection":false},{"text":"Bellwood Yard, Atlanta, GA, USA","magicKey":"dHA9MCN0dj02NTc3YTY2NiNsb2M9NDIyMTM5MSNsbmc9NTUjcGw9MzQxNjY2MSNsYnM9MTQ6OTIyMjEyMyNsbj1Xb3JsZA==","isCollection":false},{"text":"Bellwood, IL, USA","magicKey":"dHA9MCN0dj02NTc3YTY2NiNsb2M9NDQyODk3MSNsbmc9NTUjcGw9MzU5NjAwMyNsYnM9MTQ6OTIyMjAwNCNsbj1Xb3JsZA==","isCollection":false}]}
```

**Comment by CGMossa** [comment]
Right, so `#[extendr(use_try_from=true)]` has the conversion you're looking for. I can't see the regular-mode conversion available unfortunately.

**Comment by lanceferrer** [comment]
I ran into this same issue. Tried This is my use case:

```rs

#[derive(Debug, Clone, IntoDataFrameRow)]
pub struct ThetaParameterUnpacked {
    pub comment: Option<String>,
    pub lower_bound_value: Option<f64>,
    pub lower_bound_start: Option<i32>,
    pub lower_bound_end: Option<i32>,
    pub estimate_value: f64,
    pub estimate_start: i32,
    pub estimate_end: i32,
    pub upper_bound_value: Option<f64>,
    pub upper_bound_start: Option<i32>,
    pub upper_bound_end: Option<i32>,
}

#[extendr(use_try_from = true)]
pub fn get_parameters(input: &str) -> Dataframe<ThetaParameterUnpacked> {
    let theta_parameters: Vec<ThetaParameter> = get_theta_parameters_from_input(input);

    let theta_parameters_unpacked = theta_parameters
        .iter()
        .map(|x| x.unpack())
        .collect::<Vec<_>>();

    match theta_parameters_unpacked.into_dataframe() {
        Ok(dataframe) => dataframe,
        Err(err) => panic!("{}", err),
    }
}
```

**Comment by JosiahParry** [comment]
Closed by #723
